tests/test_optics.py

```diff
@@ -458,7 +479,7 @@ def test_OpticalPSF_pupil_plane():
         do_pickle(test_psf)
 
     # Make a smaller pupil plane image to test the pickling of this, even without slow tests.
-    with assert_warns(galsim.GalSimWarning):
+    with assert_warns(galsim.GalSimWarning) if not do_slow_tests else ExitStack():
```

Comment:
I don't understand the point of this.
If do_slow_tests is true, then we don't assert that there is a warning?  Instead we just use ExitStack, which isn't actually doing anything, I guess.  But either way, the next lines get called, so what's supposed to be the difference between when we are willing to do the slow tests versus not?

Reply:
When do_slow_tests is true, the pupil image is sufficiently high-res to not trigger the warning.  I suppose though that it may be better to bin by a larger factor in that case rather than elide the assert_warns...  I'll look into that.

Reply:
Changing the binning factor did the trick, so I removed the ExitStack stuff.